src/Data/OpenApi/Internal/Schema/Validation.hs

```diff
@@ -477,13 +477,20 @@ inferParamSchemaTypes sch = concat
         , has (pattern._Just) ] ]
   ]
 
+-- compute number of passing variants and variants
+countPassingVariants :: Value -> [Referenced Schema] -> Validation s (Int, Int)
+countPassingVariants val variants = do
+      res <- forM variants $ \var ->
+        True <$ validateWithSchemaRef var val <|> return False
+      pure (length $ filter id res, length variants)
+
+
 validateSchemaType :: Value -> Validation Schema ()
 validateSchemaType val = withSchema $ \sch ->
   case sch of
     (view oneOf -> Just variants) -> do
-      res <- forM variants $ \var ->
-        (True <$ validateWithSchemaRef var val) <|> (return False)
-      case length $ filter id res of
+      (npassing, _nvars) <-  countPassingVariants val variants
```

Comment:
Do you really need to compute second element in the function? Looks like it's not used anywhere 🤔

Reply:
right, I was thinking if to use the `countPassingVariants` also for the `allOf` case but I couldn't decide :-) I'm open to remove it  or to use it